webapp/components/FilterPosts/FilterPosts.vue

```diff
@@ -0,0 +1,282 @@
+                  </ds-flex-item>
+                  <ds-space margin-bottom="xx-large" />
+                </ds-flex>
+              </ds-flex>
+            </ds-flex>
+          </ds-flex-item>
+          <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+            <ds-flex v-for="category in chunk[4]" :key="category.id" class="categories-list">
+              <ds-flex>
+                <ds-flex-item width="100%">
+                  <ds-button
+                    :icon="category.icon"
+                    :primary="isActive(category.id)"
+                    @click.stop.prevent="toggleCategory(category.id)"
+                  />
+                  <ds-space margin-bottom="small" />
+                </ds-flex-item>
+                <ds-flex>
+                  <ds-flex-item>
+                    <label>{{ category.name }}</label>
+                  </ds-flex-item>
+                  <ds-space margin-bottom="xx-large" />
+                </ds-flex>
+              </ds-flex>
+            </ds-flex>
+          </ds-flex-item>
+          <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+            <ds-flex v-for="category in chunk[5]" :key="category.id" class="categories-list">
+              <ds-flex>
+                <ds-flex-item width="100%">
+                  <ds-button
+                    :icon="category.icon"
+                    :primary="isActive(category.id)"
+                    @click.stop.prevent="toggleCategory(category.id)"
+                  />
+                  <ds-space margin-bottom="small" />
+                </ds-flex-item>
+                <ds-flex>
+                  <ds-flex-item>
+                    <label>{{ category.name }}</label>
+                  </ds-flex-item>
+                  <ds-space margin-bottom="xx-large" />
+                </ds-flex>
+              </ds-flex>
+            </ds-flex>
+          </ds-flex-item>
+          <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+            <ds-flex v-for="category in chunk[6]" :key="category.id" class="categories-list">
+              <ds-flex>
+                <ds-flex-item width="100%">
+                  <ds-button
+                    :icon="category.icon"
+                    :primary="isActive(category.id)"
+                    @click.stop.prevent="toggleCategory(category.id)"
+                  />
+                  <ds-space margin-bottom="small" />
+                </ds-flex-item>
+                <ds-flex>
+                  <ds-flex-item>
+                    <label>{{ category.name }}</label>
+                  </ds-flex-item>
+                  <ds-space margin-bottom="xx-large" />
+                </ds-flex>
+              </ds-flex>
+            </ds-flex>
+          </ds-flex-item>
+          <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+            <ds-flex v-for="category in chunk[7]" :key="category.id" class="categories-list">
```

Comment:
Not happy with how many lines of code this component has just to break up the categories into 8 rows, but try as I might to iterate over the subarrays of the chunk array, I couldn't get it right... hopefully someone will be able to help me reduce this code

@roschaefer @Tirokk @ogerly

Reply:
You could wrap the chunks in a separate component and pass `chunk` as a prop and an integer as `key`. Would that help?

Reply:
yes, that would help, I believe